src/foam/lib/csv/CSVOutputter.js

```diff
@@ -0,0 +1,227 @@
+            },
+            Undefined: function(value) {},
+            Null: function(value) {}
+          }, function(value) {
+            this.outputValue_(value.toString());
+        }),
+      javaCode: `
+        if ( value instanceof String ) {
+          getSb().append("\\"");
+          getSb().append(((String)value).replace("\\"", "\\"\\""));
+          getSb().append("\\"");
+        } else if ( value instanceof Number ) {
+          getSb().append(value.toString());
+        } else if ( value instanceof Boolean ) {
+          getSb().append(value.toString());
+        } else if ( value instanceof Date ) {
+          getSb().append(value.toString());
+        } else if ( value instanceof FObject ) {
+          outputValue_(value.toString());
+        } else if ( value instanceof List ) {
+          outputValue_(value.toString());
+        } else {
+          outputValue_(value.toString());
+        }
+      `
+    },
+    {
+      name: 'newLine_',
+      code: function() {
+        this.csv += '\n';
+        this.isFirstColumn = true;
+      },
+      javaCode: `
+        getSb().append("\\n");
+        setIsFirstColumn(true);
+      `
+    },
+    {
+      name: 'toString',
+      code: function() {
+        return this.csv;
+      },
+      javaCode: 'return getSb().toString();'
+    },
+    {
+      name: 'headerOutput',
+      args: [
+        { type: 'FObject', name: 'obj' }
+      ],
+      code: function() {
+        var element;
+        this.props.forEach((name) => {
+          element = this.of.getAxiomByName(name);
+          element.toCSVLabel(this, element);
+        });
+        this.newLine_();
+        this.isFirstRow = false;
+      },
+      javaCode: `
+        Object propObj;
+        PropertyInfo columnProp;
+        String[] tableColumnNames = getProps();
+
+        for (String propName: tableColumnNames) {
+          propObj = obj.getProperty(propName);
+          columnProp = (PropertyInfo) getOf().getAxiomByName(propName);
+          columnProp.toCSVLabel(this, propObj);
+        }
+        newLine_();
+        setIsFirstRow(false);
+      `
+    },
+    {
+      name: 'outputFObject',
+      args: [
+        { type: 'FObject', name: 'obj' }
+      ],
+      code: function(obj) {
+        var element;
+
+        if ( ! this.of ) this.of = obj.cls_;
+
+        if ( this.isFirstRow ) this.headerOutput();
+
+        this.props.forEach((name) => {
+          element = this.of.getAxiomByName(name);
+          element.toCSV(x, obj, this, element);
+        });
+
+        this.newLine_();
+      },
+      javaCode: `
+        Object propObj;
+        PropertyInfo columnProp;
+        String[] tableColumnNames = getProps();
+
+        if ( ! isPropertySet("of") || getOf() == null ) setOf(obj.getClassInfo());
+
+        if ( getIsFirstRow() ) headerOutput(obj);
+
+        for (String propName : tableColumnNames) {
+          propObj = obj.getProperty(propName);
```

Comment:
Similar comment here as above. Ditch propObj and rename columnProp to axiom/prop.